### PR TITLE
CI/CD Setup/Development

### PR DESCRIPTION
**Pull request by mathieutrudeau**
- Added deployment workflow (`deploy-services.yml`) for Docker image publishing.
- Removed hardcoded endpoints for better flexibility.
- Updated Swagger documentation style and added XML comments.
- Adjusted backend configurations for Docker and improved middleware.
- Updated frontend tests to use environment variables.
- Deleted `package-lock.json` for simplified package management.

### CODE REVIEW

File: Backend/Backend/Program.cs [hunk 1]
```diff
@@ -88,36 +94,43 @@ public class Program
...
+            else
                 options.UseSqlServer(config.GetConnectionString("DefaultConnection"));
         });
 
 
         builder.Services.AddCors(o => o.AddPolicy("Dev-policy", builder =>
         {
             builder.WithOrigins("http://localhost:3000", "https://localhost:3000",
-            "http://localhost:3500", "https://localhost:3500","http://fronted:3500", "https://fronted:3500")
+            "http://localhost:3500", "https://localhost:3500", "http://fronted:3500", "https://fronted:3500",
```
Comment (BigSaladMan) [inline Backend/Backend/Program.cs:114]: Small bit of hardcode here, but this is something we can fix later

**Review by BigSaladMan — APPROVED** [review]

File: Backend/Backend/Program.cs (hunk 1 above)
Comment (mathieutrudeau) [reply Backend/Backend/Program.cs:114]: Thanks for exposing this...

**Review by Radwanbaba13 — APPROVED** [review]
Looks good

**Comment by sFranchetto** [comment]
**Overall Impression:**
The changes in this pull request are well-organized and contribute to enhancing the overall maintainability and flexibility of the project. The addition of a deployment workflow, removal of hardcoded endpoints, and updates to Swagger documentation and backend configurations demonstrate a commitment to best practices. Additionally, the adjustment of frontend tests to utilize environment variables and the removal of package-lock.json align with a streamlined and more manageable approach to package management.

**Specific Feedback:**

1. **Deployment Workflow:**
- The addition of the deployment workflow (deploy-services.yml) is a positive step towards automating the Docker image publishing process. This will streamline the deployment pipeline and ensure consistency in the deployment process.

**2. Endpoint Flexibility:**
- The removal of hardcoded endpoints is a commendable improvement. This enhances flexibility and makes it easier to adapt the application to different environments or changing backend configurations.

**3. Swagger Documentation and XML Comments:**
- Updating the Swagger documentation style and adding XML comments is a valuable contribution. This not only improves the readability of the documentation but also enhances the automatic generation of documentation, making it a valuable resource for developers.

**4. Backend Configurations and Middleware:**
The adjustments to backend configurations for Docker and improvements to middleware demonstrate a keen understanding of backend architecture. These changes contribute to better efficiency and maintainability of the backend infrastructure.

**5. Frontend Test Updates:**
- Updating frontend tests to use environment variables is a good practice. This allows for better configuration management and ensures that tests are adaptable to different environments without hardcoded values.

**6. Package Management:**
- Deleting package-lock.json for simplified package management is a decision that aligns with best practices. This file is automatically generated and can cause unnecessary conflicts. Relying on package.json alone is more straightforward and cleaner.

**Conclusion:**
These changes contribute significantly to the project's maintainability, flexibility, and overall development workflow. The attention to detail, adherence to best practices, and improvements in both backend and frontend aspects make this pull request a valuable addition to the codebase. Well done!

**Pull request merged**